README.md

```diff
@@ -9,76 +9,194 @@ Adrenaline
+          dataProps[prop] = state[query.id];
+          return Promise.resolve();
+        } else {
+          return fetch(`/api/resource/${query.id}`)
+            .then((response)=>{
+              const resource = response.json();
+              dispatch(updateCache(resource));
+              dataProps[prop] = resource;
+            })
+        }
+      }))
+      .then(() => dataProps); // resolve the data props
+  }
+}
+```
 
-### Only `id`
+### `<AdrenalineProvider getState, subscribe, dispatch>`
+Adrenaline requires a Provider (usually close to the root of your application tree), similar to the [react-redux `<Provider>`](https://github.com/rackt/react-redux/blob/master/docs/api.md#provider-store).
+The provider's purpose is to expose an api to containers (_through context_), binding
+them to a store (redux or otherwise) and an Adrenaline adapter strategy for resolving state
+and fetching data.
 
-Currently **Adrenaline** supports only `id` as a name for id attribute on your type.
+Thankfully, Adrenaline comes with a built-in redux-based provider that binds Adrenaline to the redux store. Place it after your redux provider and the store will be retrieved from context. Optional props 'stateSelector' and 'actionType' allow you to adjust the domain in the state tree that adapters
+will resolve from and restricted set of actions that containers subscribe to for triggering data refresh.
+`<AdrenalineReduxProvider [stateSelector=(state)=>state], [actionType="*"]>`
 
-```javascript
-// Invalid
-const fooType = new GraphQLObjectType({
-  name: 'Foo',
-  fields: () => {
-    customIdName: {
-      type: new GraphQLNonNull(GraphQLID),
-      description: 'Entity id',
-    },
-    baz: {
-      type: GraphQLString,
-      description: 'some stuff',
-    },
-  },
-});
-
-// Valid
-const fooType = new GraphQLObjectType({
-  name: 'Foo',
-  fields: () => {
-    id: {
-      type: new GraphQLNonNull(GraphQLID),
-      description: 'Entity id',
-    },
-    baz: {
-      type: GraphQLString,
-      description: 'some stuff',
-    },
-  },
+####Example
+Building upon the resource adapter example, we place the adrenaline's cache at a sub-domain
+in the state tree and optimize containers by ensuring they only subscribe to and re-resolve
+data props after 'UPDATE_CACHE' actions.
+```
+import { createStore, combineReducers} from 'redux';
+import { Provider } from 'react-redux';
+import { AdrenalineReduxProvider } from 'adrenaline';
+import { resourceAdapter, reducer, UPDATE_CACHE } from './util/ResourceAdapter';
+const store = combineReducers({
+  'adrenalineCache': reducer
 });
+ReactDOM.render(
+  <Provider store={store}>
+    <AdrenalineReduxProvider
+      adapter={resourceAdapter}
+      stateSelector={ (state) => state.adrenalineCache }
+      actionType={UPDATE_CACHE} >
+      { /* Your application */ }
+    </AdrenalineReduxProvider>
+  </Provider>
+  , document.body);
 ```
 
-### `id` is required
+### `createContainer(options)`
```

Comment:
you could always adopt the same terminology as Relay here. This would be a 'rootContainer'.